synapse/lib/config.py

```diff
@@ -63,14 +88,23 @@ def getJsValidator(schema, use_default=True):
     if schema.get('$schema') is None:
         schema['$schema'] = 'http://json-schema.org/draft-07/schema#'
 
+    # This handlers dictionary is used by the jsonschema validator to attempt to
+    # resolve schema '$ref' values locally from disk and will raise an exception
+    # if the schema isn't found locally
+    handlers = {
+        'http': localSchemaRefHandler,
+        'https': localSchemaRefHandler,
+    }
+
     # It is faster to hash and cache the functions here than it is to
-    # generate new functions each time we have the same schema.
+    # generate new functions each time we have the same schema. We don't have to
+    # include the handlers in the hash key because it's always used.
     key = s_hashitem.hashitem((schema, use_default))
     func = _JsValidators.get(key)
     if func:
         return func
 
-    func = fastjsonschema.compile(schema, use_default=use_default)
+    func = fastjsonschema.compile(schema, handlers=handlers, use_default=use_default)
```

Comment:
I might be misreading it, but this will raise for any schema where we don't have a local file for a ref.  Since we exposed jsonschema in storm, there could be arbitrary schemas w/refs we don't know about, that will now break?

Reply:
I didn't know about the jsonschema storm lib, thanks for pointing it out. I updated this function so it will use the local cache handlers by default but can take `handlers=None` and then updated the storm lib to pass `handlers=None`. I think that should address your concern?